File: Java/Part-1/ArmstrongNumber.java
/*
Problem: Write a Java program to check whether a given number is an Armstrong number or not.
---------------------------------------------------------------------------------------------
Example:
    Input  : 153
    Process: 1^3 + 5^3 + 3^3 = 153
    Output : Armstrong Number

Logic Used:
    - Count number of digits using String.valueOf(num).length()
    - Extract each digit using % 10
    - Raise it to power of count and sum
    - Compare sum with original number
---------------------------------------------------------------------------------------------
*/

import java.util.*;

class ArmstrongNumber
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter number:");

        int num = sc.nextInt();
        int original_num = num;
        int sum = 0;

        int count = String.valueOf(Math.abs(num)).length(); // Number of digits

        while(num > 0)
        {
            int rem = num % 10;
            sum += (int)Math.pow(rem, count); // Sum of digits^count
            num = num / 10;
        }

        if(original_num == sum)
        {
            System.out.println(original_num + " is an Armstrong number");
        }
        else
        {
            System.out.println(original_num + " is NOT an Armstrong number");
        }
    }
}

/* 
Output:-
-------
Enter number:
153
153 is an Armstrong number
*/ 

File: Python/Part-1/ArmstrongNumber.py
"""
Problem: Check whether a given number is an Armstrong number or not.
Example:
    Input  : 153
    Process: 1^3 + 5^3 + 3^3 = 153
    Output : Armstrong Number
Logic:
    - Count number of digits using len(str(num))
    - Extract each digit using % 10
    - Raise it to power of count and sum
    - Compare sum with original number
"""

num = int(input("Enter number: "))
original_num = num
count = len(str(abs(num)))
sum = 0
temp = num

while temp > 0:
    rem = temp % 10
    sum += rem ** count
    temp = temp // 10

if sum == original_num:
    print(f"{original_num} is an Armstrong number")
else:
    print(f"{original_num} is NOT an Armstrong number")



"""
Output:-
--------
Enter number: 153
153 is an Armstrong number

"""